cmd/create/execution.go

```diff
@@ -195,16 +200,24 @@ func createExecutionCommand(ctx context.Context, args []string, cmdCtx cmdCore.C
 	case Recover:
 		return recoverExecution(ctx, execParams.name, sourceProject, sourceDomain, cmdCtx, executionConfig, targetExecName)
 	case Task:
-		if executionRequest, err = createExecutionRequestForTask(ctx, execParams.name, sourceProject, sourceDomain, cmdCtx, executionConfig, targetExecName); err != nil {
+		executionRequest, err = createExecutionRequestForTask(ctx, execParams.name, sourceProject, sourceDomain, cmdCtx, executionConfig, targetExecName)
+		if err != nil {
 			return err
 		}
 	case Workflow:
-		if executionRequest, err = createExecutionRequestForWorkflow(ctx, execParams.name, sourceProject, sourceDomain, cmdCtx, executionConfig, targetExecName); err != nil {
+		executionRequest, err = createExecutionRequestForWorkflow(ctx, execParams.name, sourceProject, sourceDomain, cmdCtx, executionConfig, targetExecName)
+		if err != nil {
 			return err
 		}
 	default:
 		return fmt.Errorf("invalid execution type %v", execParams.execType)
 	}
+
+	if executionRequest.Spec.ClusterAssignment == nil {
```

Comment:
spec is always initialized in https://github.com/flyteorg/flytectl/blob/master/cmd/create/execution_util.go#L155
If its nil, it will fail on server anyway, because what's the point of sending ExecutionRequest without spec?